submissions/dashakim/oop-task/js/app.js

```diff
@@ -0,0 +1,100 @@
+//game score
+var score = 0
+//Enemy class with update(), render() methods
+var Enemy = function (x, y, speed) {
+    this.x = x
+    this.y = y
+    this.speed = speed
+    this.sprite = 'images/car.png';
+};
+var END_CANVAS = 510
+var BEGIN_CANVAS = -50
+Enemy.prototype.update = function (dt) {
+
+    this.x += this.speed * dt
+    if (this.x > END_CANVAS) {
+        this.x = BEGIN_CANVAS
+        //random speed for enemies
+        this.speed = 100 + Math.floor(Math.random() * 222)
+    }
+    // return to start position player when collide with enemy
+    if (player.x > (this.x - 60) && player.x < (this.x + 60) && player.y > (this.y - 60) && player.y < (this.y + 60)) {
+        player.startPosition();
+    }
+};
+
+Enemy.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+var allEnemies = []
+var enemyLocation = [60, 140, 220]
+
+enemyLocation.forEach(function (y) {
+    allEnemies.push(new Enemy(0, y, 200))
+})
+
+//start location for player
+var START_PLAYER_X = 200
+var START_PLAYER_Y = 404
+
+//class player
+var Player = function () {
+    this.startPosition()
+    this.player = 'images/char-cat-girl.png'
+}
+//back to start
+Player.prototype.startPosition = function () {
+    this.x = START_PLAYER_X
+    this.y = START_PLAYER_Y
+}
+
+Player.prototype.update = function () {
+    if (this.y <= 0) {
+        this.startPosition()
+        score++
+        displayStats()
```

Comment:
`score` and `displayStats` are 'guests' from the global scope (yet not constants) and also make class too specific. Not good inside the class.
You might (1) use some function like `addScoreBy(increment)` which would call `displayStats` in its turn and also (2) pass this function to the player's instance at initialization (yes, you can pass function as arguments).